clients/application-kafka/src/main/java/org/eclipse/hono/application/client/kafka/impl/KafkaBasedCommandSender.java

```diff
@@ -203,25 +203,28 @@ public KafkaBasedCommandSender(
                 correlationId,
                 COMMAND_TIMEOUT,
                 // Remove the correlation id if times out
-                x -> Optional.ofNullable(pendingCommandResponses.get(tenantId))
-                        .ifPresent(ids -> ids.remove(correlationId)),
+                x -> removeCorrelationId(tenantId, correlationId),
                 span);
 
         subscribeForCommandResponse(tenantId, span)
-                .compose(ok -> sendCommand(tenantId, deviceId, command, contentType, data, correlationId, properties,
-                        true, span.context())
-                                .onSuccess(sent -> {
-                                    //Store the correlation id and the expiring command promise
-                                    pendingCommandResponses.computeIfAbsent(tenantId, k -> new ConcurrentHashMap<>())
-                                            .put(correlationId, expiringCommandPromise);
-                                    LOGGER.debug("sent command [correlation-id: {}] and waiting for response", correlationId);
-                                    span.log("sent command and waiting for response");
-                                })
-                                .onFailure(error -> {
-                                    LOGGER.debug("error sending command", error);
-                                    TracingHelper.logError(span, "error sending command", error);
-                                    expiringCommandPromise.tryCompleteAndCancelTimer(Future.failedFuture(error));
-                                }));
+                .compose(ok -> {
+                    //Store the correlation id and the expiring command promise
+                    pendingCommandResponses.computeIfAbsent(tenantId, k -> new ConcurrentHashMap<>())
+                            .put(correlationId, expiringCommandPromise);
```

Comment:
With that, I think the extra `pendingResponsesForTenant` null/empty check in `getCommandResponseHandler` can be replaced with the `Optional.ofNullable` code like before. The original motivation of logging this case explicitly because of integration test errors is probably obsolete now.

Reply:
I was not aware that you have introduced these log statements for only the integration tests. It's good to remove it now.